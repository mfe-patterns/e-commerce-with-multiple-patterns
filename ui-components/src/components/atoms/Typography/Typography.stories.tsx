import { storiesOf } from '@storybook/react';
import * as React from 'react';
import {
  Heading1,
  Heading2,
  Heading3,
  Heading4,
  PText,
  Label,
  HelpText,
  HelpTextType
} from './Typography';
import * as S from './Typography.styles';

storiesOf('Atoms', module).add('Typography', () => (
  <>
    <S.Wrapper>
      <Heading1>Heading1</Heading1>
    </S.Wrapper>
    <S.Wrapper>
      <Heading2>Heading2</Heading2>
    </S.Wrapper>
    <S.Wrapper>
      <Heading3>Heading3</Heading3>
    </S.Wrapper>
    <S.Wrapper>
      <Heading4>Heading4</Heading4>
    </S.Wrapper>
    <S.Wrapper>
      <PText>Paragraph</PText>
    </S.Wrapper>
    <S.Wrapper>
      <Label>Label</Label>
    </S.Wrapper>
    <S.Wrapper>
      <HelpText type={HelpTextType.DEFAULT}>HelpText</HelpText>
    </S.Wrapper>
    <S.Wrapper>
      <HelpText type={HelpTextType.SUCCESS}>HelpTextError</HelpText>
    </S.Wrapper>
    <S.Wrapper>
      <HelpText type={HelpTextType.ERROR}>HelpTextSuccess</HelpText>
    </S.Wrapper>
  </>
));
