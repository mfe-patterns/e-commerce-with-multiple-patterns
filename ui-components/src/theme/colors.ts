interface Shade {
  50: string;
  100: string;
  200: string;
  300: string;
  400: string;
  500: string;
  600: string;
  700: string;
}

const dark: Shade = {
  50: "#11111193",
  100: "#111111a2",
  200: "#111111B3",
  300: "#111111B9",
  400: "#111111CC",
  500: "#111111d8",
  600: "#111111eb",
  700: "#111111",
};
const neutral: Shade = {
  50: "#fafafa",
  100: "#f2f2f2",
  200: "#ededed",
  300: "#dedede",
  400: "#cccccc",
  500: "#666666",
  600: "#434343",
  700: "#343434",
};

const primary: Shade = {
  50: "#e3f2fd",
  100: "#bbdefb",
  200: "#90caf9",
  300: "#64b5f6",
  400: "#42a5f5",
  500: "#2196f3",
  600: "#1e88e5",
  700: "#1976d2",
};

const white = "#ffffff";

export interface IColorPallete {
  neutral: Shade;
  dark: Shade;
  primary: Shade;
  white: string;
  black: string;
  error: string;
  success: string;
  text: string;
  transparent: string;
  link: string;
}

export const colors: IColorPallete = {
  neutral,
  dark,
  primary,
  white,
  black: "#000000",
  error: "#dc3545",
  success: "#28a745",
  text: dark[700],
  transparent: "transparent",
  link: "#0064cc",
};
