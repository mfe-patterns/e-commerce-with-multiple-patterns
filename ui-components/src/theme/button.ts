import { colors } from "./colors";

interface IButtonType {
  color: string;
  backgroundColor?: string;
  borderColor?: string;
  fontFamily: string;
  border: string;
  disabled: {
    color: string;
    backgroundColor?: string;
    borderColor?: string;
  };
  horizontalPadding: boolean;
}

export const primary: IButtonType = {
  color: colors.white,
  borderColor: colors.transparent,
  fontFamily: "Roboto-Regular",
  border: "none",
  disabled: {
    color: colors.white,
    borderColor: colors.transparent,
  },
  horizontalPadding: true,
};
export const secondary: IButtonType = {
  backgroundColor: colors.white,
  color: colors.text,
  fontFamily: "Roboto-Regular",
  border: "1px solid",
  disabled: {
    backgroundColor: colors.white,
    color: colors.dark[100],
  },
  horizontalPadding: true,
};

interface IButtonSize {
  height: string;
  borderRadius: string;
  fontSize: string;
  lineHeight: string;
  letterSpacing: string;
  paddingVertical: string;
  paddingHorizontal: string;
}

interface ISize {
  xlarge: IButtonSize;
  large: IButtonSize;
  medium: IButtonSize;
  small: IButtonSize;
  xsmall: IButtonSize;
}

const size: ISize = {
  xlarge: {
    height: "56px",
    borderRadius: "28px",
    fontSize: "16px",
    lineHeight: "24px",
    letterSpacing: "0.7px",
    paddingVertical: "15px",
    paddingHorizontal: "29.5px",
  },
  large: {
    height: "46px",
    borderRadius: "23px",
    fontSize: "15px",
    lineHeight: "20px",
    letterSpacing: "0.7px",
    paddingVertical: "12px",
    paddingHorizontal: "25.5px",
  },
  medium: {
    height: "40px",
    borderRadius: "20px",
    fontSize: "14px",
    lineHeight: "20px",
    letterSpacing: "0.7px",
    paddingVertical: "10px",
    paddingHorizontal: "21.5px",
  },
  small: {
    height: "36px",
    borderRadius: "18px",
    fontSize: "13px",
    lineHeight: "19px",
    letterSpacing: "0.7px",
    paddingVertical: "8px",
    paddingHorizontal: "20px",
  },
  xsmall: {
    height: "29px",
    borderRadius: "14.5px",
    fontSize: "12px",
    lineHeight: "19px",
    letterSpacing: "0.7px",
    paddingVertical: "5px",
    paddingHorizontal: "14px",
  },
};

export interface IButton {
  type: {
    primary: IButtonType;
    secondary: IButtonType;
  },
  size: ISize;
}

export const button: IButton = {
  type: {
    primary: {
      ...primary,
      disabled: {
        ...primary.disabled,
        backgroundColor: colors.primary[200],
      },
      backgroundColor: colors.primary[500],
    },
    secondary: {
      ...secondary,
      borderColor: colors.primary[500],
      disabled: {
        ...secondary.disabled,
        borderColor: colors.primary[200],
      },
    },
  },
  size,
};
