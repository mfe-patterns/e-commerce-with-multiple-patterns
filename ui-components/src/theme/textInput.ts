import { colors } from './colors';
import { FontFamily } from './typography';

interface ITextInputType {
  color: string;
  placeholderColor?: string;
  backgroundColor: string;
  borderColor: string;
}

export interface ITextInput {
  fontFamily: FontFamily;
  default: ITextInputType;
  disabled: ITextInputType;
  error: ITextInputType;
  success: ITextInputType;
}

export const textInput: ITextInput = {
  fontFamily: 'Roboto-Regular',
  default: {
    color: colors.text,
    placeholderColor: colors.dark[100],
    backgroundColor: colors.white,
    borderColor: colors.neutral[300],
  },
  disabled: {
    color: colors.dark[50],
    backgroundColor: colors.neutral[50],
    borderColor: colors.neutral[300],
  },
  error: {
    color: colors.text,
    backgroundColor: colors.white,
    borderColor: colors.error,
  },
  success: {
    color: colors.text,
    backgroundColor: colors.white,
    borderColor: colors.success,
  },
};
