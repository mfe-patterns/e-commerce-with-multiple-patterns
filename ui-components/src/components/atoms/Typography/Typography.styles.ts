import styled from "styled-components";

import { theme as coreTheme } from "../../../theme/theme";

export interface IText {
  color?: string;
  theme: typeof coreTheme;
}

export const Wrapper = styled.div`
  margin: 20px;
`;

export const Heading1 = styled.h1<IText>`
  font-family: ${({ theme }) => theme.typography.heading1.fontFamily};
  font-size: ${({ theme }) => theme.typography.heading1.fontSize};
  line-height: ${({ theme }) => theme.typography.heading1.lineHeight};
  letter-spacing: ${({ theme }) => theme.typography.heading1.letterSpacing};
  color: ${({ color, theme }) => color || theme.colors.text};
`;

export const Heading2 = styled.h2<IText>`
  font-family: ${({ theme }) => theme.typography.heading2.fontFamily};
  font-size: ${({ theme }) => theme.typography.heading2.fontSize};
  line-height: ${({ theme }) => theme.typography.heading3.lineHeight};
  letter-spacing: ${({ theme }) => theme.typography.heading2.letterSpacing};
  color: ${({ color, theme }) => color || theme.colors.text};
`;

export const Heading3 = styled.h3<IText>`
  font-family: ${({ theme }) => theme.typography.heading3.fontFamily};
  font-size: ${({ theme }) => theme.typography.heading3.fontSize};
  line-height: ${({ theme }) => theme.typography.heading3.lineHeight};
  letter-spacing: ${({ theme }) => theme.typography.heading3.letterSpacing};
  color: ${({ color, theme }: IText) => color || theme.colors.text};
`;

export const Heading4 = styled.h4<IText>`
  font-family: ${({ theme }) => theme.typography.heading4.fontFamily};
  font-size: ${({ theme }) => theme.typography.heading4.fontSize};
  line-height: ${({ theme }) => theme.typography.heading4.lineHeight};
  letter-spacing: ${({ theme }) => theme.typography.heading4.letterSpacing};
  color: ${({ color, theme }: IText) => color || theme.colors.text};
`;

export const Heading5 = styled.h5<IText>`
  font-family: ${({ theme }) => theme.typography.heading5.fontFamily};
  font-size: ${({ theme }) => theme.typography.heading5.fontSize};
  line-height: ${({ theme }) => theme.typography.heading5.lineHeight};
  letter-spacing: ${({ theme }) => theme.typography.heading5.letterSpacing};
  color: ${({ color, theme }: IText) => color || theme.colors.text};
`;

export const PText = styled.p<IText>`
  font-family: ${({ theme }) => theme.typography.pText.fontFamily};
  font-size: ${({ theme }) => theme.typography.pText.fontSize};
  line-height: ${({ theme }) => theme.typography.pText.lineHeight};
  letter-spacing: ${({ theme }) => theme.typography.pText.letterSpacing};
  color: ${({ color, theme }: IText) => color || theme.colors.text};
`;

export const Label = styled.label<IText>`
  font-family: ${({ theme }) => theme.typography.label.fontFamily};
  font-size: ${({ theme }) => theme.typography.label.fontSize};
  line-height: ${({ theme }) => theme.typography.label.lineHeight};
  letter-spacing: ${({ theme }) => theme.typography.label.letterSpacing};
  color: ${({ color, theme }: IText) => color || theme.colors.text};
`;

export const HelpText = styled.p<IText>`
  font-family: ${({ theme }) => theme.typography.helpText.fontFamily};
  font-size: ${({ theme }) => theme.typography.helpText.fontSize};
  line-height: ${({ theme }) => theme.typography.helpText.lineHeight};
  letter-spacing: ${({ theme }) => theme.typography.helpText.letterSpacing};
  color: ${({ color }: IText) => color};
`;
