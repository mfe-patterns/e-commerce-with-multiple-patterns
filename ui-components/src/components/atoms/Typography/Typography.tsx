import React, { ReactNode } from 'react';
import * as S from './Typography.styles';
import { theme } from '../../../theme/theme';

export interface IText {
  children: ReactNode | string;
  color?: string;
}

export enum HelpTextType {
  DEFAULT = 'default',
  SUCCESS = 'success',
  ERROR = 'error',
}

export interface IHelpText extends IText {
  type?: HelpTextType.SUCCESS | HelpTextType.ERROR | HelpTextType.DEFAULT;
}

export const Heading1: React.FC<IText> = ({ children, ...others }: IText) => {
  return <S.Heading1 {...others}>{children}</S.Heading1>;
};

export const Heading2: React.FC<IText> = ({ children, ...others }: IText) => {
  return <S.Heading2 {...others}>{children}</S.Heading2>;
};

export const Heading3: React.FC<IText> = ({ children, ...others }: IText) => {
  return <S.Heading3 {...others}>{children}</S.Heading3>;
};

export const Heading4: React.FC<IText> = ({ children, ...others }: IText) => {
  return <S.Heading4 {...others}>{children}</S.Heading4>;
};

export const PText: React.FC<IText> = ({ children, ...others }: IText) => {
  return <S.PText {...others}>{children}</S.PText>;
};

export const Label: React.FC<IText> = ({ children, ...others }: IText) => {
  return <S.Label {...others}>{children}</S.Label>;
};

export const HelpText: React.FC<IHelpText> = ({
  children,
  type = HelpTextType.DEFAULT,
  ...others
}: IHelpText) => {
  const color = {
    default: theme.colors.dark[400],
    success: theme.colors.success,
    error: theme.colors.error,
  };
  return (
    <S.HelpText {...others} color={color[type]}>
      {children}
    </S.HelpText>
  );
};