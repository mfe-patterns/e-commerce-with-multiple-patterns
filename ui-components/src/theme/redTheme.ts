import { ITypography, typography } from "./typography";
import { IColorPallete, colors } from "./colors";
import { ITextInput, textInput } from "./textInput";
import { button, IButton, primary, secondary } from "./button";

const redPalette: IColorPallete = {
  ...colors,
  primary: {
    50: "#ffebee",
    100: "#ffcdd2",
    200: "#ef9a9a",
    300: "#e57373",
    400: "#ef5350",
    500: "#f44336",
    600: "#e53935",
    700: "#d32f2f",
  },
};

const redButton: IButton = {
  ...button,
  type: {
    primary: {
      ...primary,
      disabled: {
        ...primary.disabled,
        backgroundColor: redPalette.primary[200],
      },
      backgroundColor: redPalette.primary[500],
    },
    secondary: {
      ...secondary,
      borderColor: redPalette.primary[500],
      disabled: {
        ...secondary.disabled,
        borderColor: redPalette.primary[200],
      },
    },
  },
};

interface ITheme {
  typography: ITypography;
  colors: IColorPallete;
  textInput: ITextInput;
  button: IButton;
  gridSize: number;
}

export const theme: ITheme = {
  typography,
  colors: redPalette,
  textInput,
  button: redButton,
  gridSize: 8,
};
