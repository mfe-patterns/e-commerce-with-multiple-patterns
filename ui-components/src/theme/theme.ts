import { ITypography, typography } from "./typography";
import { IColorPallete, colors } from "./colors";
import { ITextInput, textInput } from "./textInput";
import { button, IButton } from "./button";

interface ITheme {
  typography: ITypography;
  colors: IColorPallete;
  textInput: ITextInput;
  button: IButton;
  gridSize: number;
}

export const theme: ITheme = {
  typography,
  colors,
  textInput,
  button,
  gridSize: 8
};
