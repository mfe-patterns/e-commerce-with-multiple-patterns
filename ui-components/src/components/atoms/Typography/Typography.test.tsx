import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';
import { ThemeProvider } from 'styled-components';

import { theme } from '../../../theme/theme';
import {
  Heading1,
  Heading2,
  Heading3,
  Heading4,
  HelpText,
  Label,
  PText,
} from './Typography';
import { HelpTextType } from './Typography';

describe('Typography', () => {
  const typographies = [
    {
      Component: Heading1,
      fontFamily: 'Roboto-Regular',
      color: '#111111',
    },
    {
      Component: Heading2,
      fontFamily: 'Roboto-Regular',
      color: '#111111',
    },
    {
      Component: Heading3,
      fontFamily: 'Roboto-Regular',
      color: '#111111',
    },
    {
      Component: Heading4,
      fontFamily: 'Roboto-Regular',
      color: '#111111',
    },
    {
      Component: PText,
      fontFamily: 'Roboto-Regular',
      color: '#111111',
    },    
    {
      Component: Label,
      fontFamily: 'Roboto-Regular',
      color: '#111111',
    },
    {
      Component: HelpText,
      fontFamily: 'Roboto-Regular',
      color: '#111111CC',
      props: { type: HelpTextType.DEFAULT },
    },
    {
      Component: HelpText,
      fontFamily: 'Roboto-Regular',
      color: '#28a745',
      props: { type: HelpTextType.SUCCESS },
    },
    {
      Component: HelpText,
      fontFamily: 'Roboto-Regular',
      color: '#dc3545',
      props: { type: HelpTextType.ERROR },
    },
  ];
  test.each(typographies)(
    'Should render typography component %p with default color',
    ({ Component, fontFamily, color, props }) => {
      const { container, getByText } = render(
        <ThemeProvider theme={theme}>
          <Component {...props}>Title</Component>
        </ThemeProvider>,
      );
      expect(container).toMatchSnapshot();
      expect(getByText('Title')).toHaveStyleRule('color', color);
      expect(getByText('Title')).toHaveStyleRule(
        'font-family',
        fontFamily,
      );
    },
  );
});
