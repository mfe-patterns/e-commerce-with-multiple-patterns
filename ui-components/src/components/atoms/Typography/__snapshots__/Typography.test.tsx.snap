// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography Should render typography component {"Component": [Function Heading1], "color": "#111111", "fontFamily": "Roboto-Regular"} with default color 1`] = `
.c0 {
  font-family: Roboto-Regular;
  font-size: 28px;
  line-height: 35px;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  color: #111111;
}

<div>
  <h1
    class="c0"
  >
    Title
  </h1>
</div>
`;

exports[`Typography Should render typography component {"Component": [Function Heading2], "color": "#111111", "fontFamily": "Roboto-Regular"} with default color 1`] = `
.c0 {
  font-family: Roboto-Regular;
  font-size: 25px;
  line-height: 30px;
  -webkit-letter-spacing: 0.3px;
  -moz-letter-spacing: 0.3px;
  -ms-letter-spacing: 0.3px;
  letter-spacing: 0.3px;
  color: #111111;
}

<div>
  <h2
    class="c0"
  >
    Title
  </h2>
</div>
`;

exports[`Typography Should render typography component {"Component": [Function Heading3], "color": "#111111", "fontFamily": "Roboto-Regular"} with default color 1`] = `
.c0 {
  font-family: Roboto-Regular;
  font-size: 22px;
  line-height: 30px;
  -webkit-letter-spacing: 0.4px;
  -moz-letter-spacing: 0.4px;
  -ms-letter-spacing: 0.4px;
  letter-spacing: 0.4px;
  color: #111111;
}

<div>
  <h3
    class="c0"
  >
    Title
  </h3>
</div>
`;

exports[`Typography Should render typography component {"Component": [Function Heading4], "color": "#111111", "fontFamily": "Roboto-Regular"} with default color 1`] = `
.c0 {
  font-family: Roboto-Regular;
  font-size: 20px;
  line-height: 27px;
  -webkit-letter-spacing: 1.2px;
  -moz-letter-spacing: 1.2px;
  -ms-letter-spacing: 1.2px;
  letter-spacing: 1.2px;
  color: #111111;
}

<div>
  <h4
    class="c0"
  >
    Title
  </h4>
</div>
`;

exports[`Typography Should render typography component {"Component": [Function HelpText], "color": "#28a745", "fontFamily": "Roboto-Regular", "props": [Object]} with default color 1`] = `
.c0 {
  font-family: Roboto-Regular;
  font-size: 11px;
  line-height: 16px;
  -webkit-letter-spacing: 0.7px;
  -moz-letter-spacing: 0.7px;
  -ms-letter-spacing: 0.7px;
  letter-spacing: 0.7px;
  color: #28a745;
}

<div>
  <p
    class="c0"
    color="#28a745"
  >
    Title
  </p>
</div>
`;

exports[`Typography Should render typography component {"Component": [Function HelpText], "color": "#111111CC", "fontFamily": "Roboto-Regular", "props": [Object]} with default color 1`] = `
.c0 {
  font-family: Roboto-Regular;
  font-size: 11px;
  line-height: 16px;
  -webkit-letter-spacing: 0.7px;
  -moz-letter-spacing: 0.7px;
  -ms-letter-spacing: 0.7px;
  letter-spacing: 0.7px;
  color: #111111CC;
}

<div>
  <p
    class="c0"
    color="#111111CC"
  >
    Title
  </p>
</div>
`;

exports[`Typography Should render typography component {"Component": [Function HelpText], "color": "#dc3545", "fontFamily": "Roboto-Regular", "props": [Object]} with default color 1`] = `
.c0 {
  font-family: Roboto-Regular;
  font-size: 11px;
  line-height: 16px;
  -webkit-letter-spacing: 0.7px;
  -moz-letter-spacing: 0.7px;
  -ms-letter-spacing: 0.7px;
  letter-spacing: 0.7px;
  color: #dc3545;
}

<div>
  <p
    class="c0"
    color="#dc3545"
  >
    Title
  </p>
</div>
`;

exports[`Typography Should render typography component {"Component": [Function Label], "color": "#111111", "fontFamily": "Roboto-Regular"} with default color 1`] = `
.c0 {
  font-family: Roboto-Regular;
  font-size: 14px;
  line-height: 21px;
  -webkit-letter-spacing: 0.7px;
  -moz-letter-spacing: 0.7px;
  -ms-letter-spacing: 0.7px;
  letter-spacing: 0.7px;
  color: #111111;
}

<div>
  <label
    class="c0"
  >
    Title
  </label>
</div>
`;

exports[`Typography Should render typography component {"Component": [Function PText], "color": "#111111", "fontFamily": "Roboto-Regular"} with default color 1`] = `
.c0 {
  font-family: Roboto-Regular;
  font-size: 14px;
  line-height: 23px;
  -webkit-letter-spacing: 0.7px;
  -moz-letter-spacing: 0.7px;
  -ms-letter-spacing: 0.7px;
  letter-spacing: 0.7px;
  color: #111111;
}

<div>
  <p
    class="c0"
  >
    Title
  </p>
</div>
`;
